kedro/config/omegaconf_config.py

```diff
@@ -0,0 +1,252 @@
+            "parameters": ["parameters*", "parameters*/**", "**/parameters*"],
+            "credentials": ["credentials*", "credentials*/**", "**/credentials*"],
+            "logging": ["logging*", "logging*/**", "**/logging*"],
+        }
+        self.config_patterns.update(config_patterns or {})
+
+        super().__init__(
+            conf_source=conf_source,
+            env=env,
+            runtime_params=runtime_params,
+        )
+
+    def __getitem__(self, key) -> Dict[str, Any]:
+        """Get configuration files by key, load and merge them, and
+        return them in the form of a config dictionary.
+
+        Args:
+            key: Key of the configuration type to fetch.
+
+        Raises:
+            KeyError: If key provided isn't present in the config_patterns of this
+               OmegaConfLoader instance.
+            MissingConfigException: If no configuration files exist matching the patterns
+                mapped to the provided key.
+
+        Returns:
+            Dict[str, Any]:  A Python dictionary with the combined
+               configuration from all configuration files.
+        """
+
+        # In the first iteration of the OmegaConfLoader we'll keep the resolver turned-off.
+        # It's easier to introduce them step by step, but removing them would be a breaking change.
+        _clear_omegaconf_resolvers()
+
+        # Load base env config
+        base_path = str(Path(self.conf_source) / self.base_env)
+        try:
+            base_config = load_and_merge_dir_config(
+                base_path, [*self.config_patterns[key]]
+            )
+            config = base_config
+        except KeyError as exc:
+            raise KeyError("Key not found in patterns") from exc
+
+        # Load chosen env config
+        run_env = self.env or self.default_run_env
+        env_path = str(Path(self.conf_source) / run_env)
+        env_config = load_and_merge_dir_config(env_path, [*self.config_patterns[key]])
+
+        # Destructively merge the two env dirs. The chosen env will override base.
+        common_keys = config.keys() & env_config.keys()
+        if common_keys:
+            sorted_keys = ", ".join(sorted(common_keys))
+            msg = (
+                "Config from path '%s' will override the following "
+                "existing top-level config keys: %s"
+            )
+            _config_logger.info(msg, env_path, sorted_keys)
+
+        config.update(env_config)
+
+        if not config:
+            raise MissingConfigException(
+                f"No files of YAML or JSON format found in {base_path} or {env_path} matching"
+                f" the glob pattern(s): {[*self.config_patterns[key]]}"
+            )
+        return config
+
+    def __repr__(self):  # pragma: no cover
+        return (
+            f"OmegaConfLoader(conf_source={self.conf_source}, env={self.env}, "
+            f"config_patterns={self.config_patterns})"
+        )
```

Comment:
Slightly confused about what these strings are for.

Reply:
The `__repr__` method in Python is used to represent a Class object as a string (https://www.educative.io/answers/what-is-the-repr-method-in-python). It's useful for debugging and in the interactive ipython mode. 

I added the `__repr__` method for `ConfigLoader` and `TemplatedConfigLoader` in https://github.com/kedro-org/kedro/pull/1966 after @noklam pointed out they're not printed properly when using in interactive mode (see https://github.com/kedro-org/kedro/pull/1966#discussion_r1004477060). The one here in `OmegaConfLoader` is basically the same as for the other config loader classes.